pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -243,10 +248,62 @@ stdenv.mkDerivation (finalAttrs: {
     ln -s $out/bin/qemu-system-${stdenv.hostPlatform.qemuArch} $out/bin/qemu-kvm
   '';
 
+  # Normal qemu package does not have installcheck make target.
+  # toolsOnly has custom check.
+  doInstallCheck = toolsOnly;
+
+  installCheckPhase =
+    if toolsOnly
+    then ''
+      runHook preInstallCheck
+
+      # Check the bin directory for unexpected files.
+      (
+        cd $out/bin;
+        echo binaries in qemu-utils package:
+        ls -al
+        for bin in *; do
+          case $bin in
+            elf2dmp) ;;
+            qemu-edid) ;;
+            qemu-img) ;;
+            qemu-io) ;;
+            qemu-nbd) ;;
+            qemu-pr-helper) ;;
+            qemu-storage-daemon) ;;
+            qemu-kvm)
+              if test -L $bin && ! test -e $bin; then
+                echo "removing broken $bin symlink"
+                echo "upstream build system should not write it, but it does, and it's not a big deal"
+                rm $bin
+              else
+                echo "$bin is not a broken symlink, which is, oddly enough, unexpected"
+                echo "make sure the build only produced the tools and nothing else"
+              fi
+              ;;
+            qemu-system-*) # custom message, but just a special case of *)
+              echo "unexpected binary in qemu-utils package: $bin"
+              echo "qemu-utils is not supposed to contain the emulator"
+              exit 1
+              ;;
```

Comment:
This seems overly defensive to me.  We're explicitly telling it not to build any system emulators.  I don't think we need to second guess the build system.

Reply:
Removed the special case, but not the entire check. I think it's good to control which binaries are in the package.

Reply:
I'm not sure — given that we're happy including everything that's currently there, I think it's very likely we'll allow everything in the future anyway.

Reply:
Ok. I've removed the check.